Fish 10526 improving concurrent implementation of corba to use lock api payara6

breakponchito opened the pull request:
This is to improve the current implementation to instead of using synchronized blocks to prevent concurrent threads to access resources, now use lock api. We decided to improve because with synchronized it is hard to determine which thread access and release the resources. Here I list the advantages to use the lock api:

1. Fine-Grained control: We are doing explicit locking and unlocking, this is more precise and efficient than the implementation with synchronized
2. Use of conditions variables: This allow threads to wait for certain conditions to be met before proceeding
3. Multiple Object locking: lock can be shared among multiple class instances, allowing you to manage synchronization across
these instances more flexible

For the tests I created the following Remote EJB:
[remoteview.zip](https://github.com/user-attachments/files/19060987/remoteview.zip)

To use you need to deploy on the server and include this jar on an standalone java application, here included also:
[EJBRemoteClient6.zip](https://github.com/user-attachments/files/19060999/EJBRemoteClient6.zip)

On the Main class you will see the remote call configuration to execute the tests with multiple threads creating 300 requests to the server. Something important for the standalone application is to indicate the correct path for the modules folder and resolve all dependencies for the EJB Client call configuration:

![image](https://github.com/user-attachments/assets/dfe77ac6-207d-4954-929d-ac8a584c920f)

to enable the new implementation with locks you need to set the following property on the system:

`com.sun.corba.ee.protocol.enablingNewFragmentProcess=true`

by default this is set as false to use the synchonized mechanism

also you can set the timeout time of the internal management of the process to wait for information on the queue with the following property:

`com.sun.corba.ee.protocol.newFragmentEmptyConditionTimeout=10000`

by default this is set with 10000 milliseconds, the unit for the value is milliseconds and this value is only used when the previous property is set as true

after doing multiple calls using the reproducer now I can't see any affectation on the threads, now all of them are finishing and also the performance is not affected:

![image](https://github.com/user-attachments/assets/bbd3ca49-2a60-4698-8299-7ebac2e18a1c)


I used the following environment for testing: Payara 6, azul JDK 11, windows 11 and maven 3.8.6

breakponchito committed:
FISH-10526: adding logs to identify deadlock

breakponchito committed:
FISH-10526: adding more thread information to identify deadlock

breakponchito committed:
FISH-10526: adding requestId for fragment processing on logs

breakponchito committed:
FISH-10526: just reordering message on logs

breakponchito committed:
FISH-10526: removing queue reference from logs to not cause ConcurrentModificationException

breakponchito committed:
FISH-10526: improving code to use lock api instead of synchronized block

breakponchito committed:
FISH-10526: adding property for timeout and code improvements based on comments

breakponchito committed:
FISH-10526: attending comments from PR

breakponchito committed:
FISH-10526: more fixes from comments on PR

breakponchito committed:
FISH-10526: adding missed copyright

breakponchito committed:
FISH-10526: preparing for next release

Pandrex247 commented on orbmain/src/main/java/com/sun/corba/ee/spi/misc/ORBConstants.java:
```diff
@@ -616,6 +618,10 @@ public static int makePersistent( int scid )
         + "ORBGmbalRootParentName" ;
+    
+    //protocol constants to enable new lock api mechanism to process fragments 
+    public static final String ENABLING_NEW_FRAGMENT_CONCURRENCY_PROCESS = SUN_PREFIX + "protocol.enablingNewFragmentProcess";
```
Final niggle - 'enable' makes more sense here than 'enabling'

```suggestion
    public static final String ENABLE_NEW_FRAGMENT_CONCURRENCY_PROCESS = SUN_PREFIX + "protocol.enableNewFragmentProcess";
```

Pandrex247 committed:
Adjust property name

'Enable' makes more sense than 'enabling'

Pandrex247 committed:
Revert "FISH-10526: preparing for next release"

This reverts commit 062f91f6d3465c7030a7ab7d609a9bdc7581b16d.

Pandrex247 committed:
glassfish-corba-4.2.4.payara-p3 release

Pandrex247 committed:
Increment version for development

Pandrex247 reviewed (changes requested):
Missing release commit

Pandrex247 commented:
https://payara.atlassian.net/wiki/spaces/PAYAR/pages/150601767/Patching+External+Projects+Process

Pandrex247 committed:
Adjust property name usage

Missed these when renaming the property name in ccefd54a525aeb7db07f1b2784ab99d03c13aa42

Pandrex247 committed:
Add patched project release profile

Pandrex247 committed:
Revert "Increment version for development"

This reverts commit d606dde8d34090b73afbcf348db75b58d0efab42.

Pandrex247 committed:
Revert "glassfish-corba-4.2.4.payara-p3 release"

This reverts commit 2d41b917a01471c80efba282e38b334b09342b34.

Pandrex247 committed:
Reapply "glassfish-corba-4.2.4.payara-p3 release"

This reverts commit ccb5d9ccd6136e3dc46c1706db1c19c107b53391.

Pandrex247 committed:
Reapply "Increment version for development"

This reverts commit 173caf740a72c37a5c629e808874efa1afc33b79.
